GetJobber/atlantis
# fix(components): fix datepicker validations

## Motivations
This fixes an `InputDate` bug that causes the field to be marked as invalid when it shouldn't. E.g. when it starts empty and is required, picking a date from the DatePicker will still mark the field as invalid even though it fills in a date.

Repro Steps: 
1. Start with an `InputDate` initialized to have no value, but with a validation requiring the field (See Testing below for a validation example). 
2. Select the `InputDate` to launch the `DatePicker` and select a date.
- **Expected:**
The date should fill the `InputDate` 
- **Observed:**
The date is filled in, but the invalid styling is applied and the validation error is shown.

https://user-images.githubusercontent.com/6612596/230157835-9fe66223-b041-496f-b943-bfd9b766d25d.mp4


## Changes



### Fixed

- Fixed an issue that was causing the InputDate field to mark the field as invalid when a date was selected from the DatePicker.

## Testing

Full tests tbd. 
1. Load storybook canvas for `InputDate`
3. Add the following validation 
```
{
    "required": {
      "message": "Start time required",
      "value": true
    }
  }
```
4. Clear the date field. 
5. OBSERVE: The proper validation error message appears, because the field is empty but required.
6. Click the InputDate to launch the DatePicker and select a date.
7. OBSERVE: The validation error message disappears correctly.
8. Click the InputDate to launch the DatePicker and select another date.
9. OBSERVE: The validation error message does not appear. 
10. Clear the date field. 
11. OBSERVE: The proper validation error message appears, because the field is empty but required.

---

[In Atlantis we use Github's built in pull request reviews](https://help.github.com/en/articles/about-pull-request-reviews).

![Random photo of Atlantis](https://loremflickr.com/672/400/atlantis)
